backend/src/api/paths/submissions.ts

```diff
@@ -1,8 +1,25 @@
 import { createRoute, OpenAPIHono, z } from "@hono/zod-openapi"
+import { PrismaClient } from "@prisma/client"
 
-import { Submission } from "../components/schemas"
+import { Submission, SubmissionStatus, TestResult } from "../components/schemas"
 
 const app = new OpenAPIHono()
+const prisma = new PrismaClient()
+
+type SubmissionStatusType = z.infer<typeof SubmissionStatus>
+type TestStatusType = "Failed" | "Passed"
+
+const isValidSubmissionStatus = (
+  status: string,
+): status is SubmissionStatusType => {
+  return ["Accepted", "CompileError", "RuntimeError", "WrongAnswer"].includes(
+    status as SubmissionStatusType,
+  )
```

Comment:
IMO: 後で`SubmissionStatusType`型の内容を変更した場合に型エラーが起きるように、配列の型が`SubmissionStatus[]`であるか確認した方が型安全で良いかなと思いました

```suggestion
  return (
    [
      "Accepted",
      "CompileError",
      "RuntimeError",
      "WrongAnswer",
    ] satisfies SubmissionStatusType[]
  ).includes(status as SubmissionStatusType)
```